src/main/java/fshop/db/FoodDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package fshop.db;
+
+import fshop.model.Food;
+import fshop.service.FoodService;
+import fshop.service.workfile.ReadCsv;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class FoodDaoImpl implements FoodDao {
+    private static final String BALANCE = "b";
+    private static final String COMA = ",";
+    private Map<Food, Integer> dataOfBalance;
+    private ReadCsv readCsv;
+    private FoodService foodService;
+
+    public FoodDaoImpl(ReadCsv readCsv) {
+        Objects.requireNonNull(readCsv);
+        this.readCsv = readCsv;
+        dataOfBalance = new HashMap<>();
+        foodService = new FoodService();
+    }
+
+    @Override
+    public void addAll() {
+        Iterator<String> listAfterRead = readCsv.read().iterator();
```

Comment:
method addl() should be parametrazide by List smth, and Storage.data = smth

Reply:
`readCsv.read()` Let's do this only once in main method and pass the result to addAll() and to updateAll() method